### ISSUE TITLE
Linux: XB1 controller rumble doesn't work unless initialized with fftest

### ISSUE THREAD

**Issue by auxym** [libretro/RetroArch#6539]
First and foremost, with 191 repos under libretro, I wasn't sure where to post this issue, so please forgive me (and point me in the right direction) if this isn't the correct place.

## Description

I own a wired xbox one controller, a third party device made by PDP. If I start Retroarch after a boot and start an N64 game with rumble support (Goldeneye, THPS2, OOT...), rumble does not work.

However, I have found a workaround: if I **first** run `fftest /dev/by-id/<event device>` and rumble it (effect 4 or 5), **then** I can start up retroarch, load up a core/ROM and rumble will work!

This leads me to believe that retroarch should be sending some sort of "initalization" stuff to the controller and isn't doing it.

### Expected behavior

Boot -> Plug in controller -> Start up retroarch -> Load core/ROM -> Working rumble

### Actual behavior

Game starts and says a rumble pack is detected, but the controller doesn't actually vibrate when it should.

### Steps to reproduce the bug

1. Cold boot PC
2. Start up retroarch and load mupen64 core
2. (I'm using SDL2 as input driver, let me know if I should test another driver)
3. Load a ROM with rumble support, say THPS2
4. Start a single session game and observe that the controller does not rumble, eg when grinding.

### Bisect Results

I'm not sure whether this is a regression, I've recently started using libretro with this hardware. If anyone has reason to believe this worked in the past, please give a working version/commit and I try testing / bisecting.

### Version/Commit

Everything installed from stable PPAs:

```
libretro-core-info/xenial,xenial,now 1.7.0-r201802192247-7fac334-12~ubuntu16.04.1 all [installed,automatic]
libretro-mupen64plus/xenial,now 2.5-r201710241904-4b41ada-12~ubuntu16.04.1 amd64 [installed]
retroarch/xenial,now 1.7.1-r201802200210-6fc6bfb-74~ubuntu16.04.1 amd64 [installed,automatic]
retroarch-assets/xenial,xenial,now 1.7.0-r201801300004-505ad62-13~ubuntu16.04.1 all [installed,automatic]
```

### Environment information

- OS: Linux Mint Sylvia with a kernel compiled from source (to get the latest xpad patches): kernel v4.15.7

**Comment by auxym** [comment]
OK, just tried `udev` joypad driver, same behavior.

I am indeed using xpad.

I can get rumble working in `fftest`, or even in retroarch/mupen64 core after running `fftest` at least once. I had at the [fftest code](https://github.com/flosse/linuxconsole/blob/master/utils/fftest.c), and it's basically doing:

1. Upload the effects with an `ioctl` syscall.
2. Play the effect by writing a struct to the device.
3. Stop by writing another struct to the device.

My hypothesis is that step 1 might be missing in retroarch. I'm looking into building a minimal version of that program that only does step 1, to validate that.

**Comment by orbea** [comment]
fwiw, my experiences with using xpad and third party controllers has been less than optimal too. This includes a broken rumble, although not quite the same as what you are experiencing.

**Comment by auxym** [comment]
Yeah, xpad is not perfect for sure. XB1 controller support seems particularly hack-ish at the moment (the driver keeps a list of quirks for each controller type).

That said, the fact that fftest works without a hitch was, to me, a good pointer that the bug might be on the retroarch side.

Do you have any idea where the linux udev rumble api would be implemented in retroarch? I'm really not much of a C hacker, but I could have a quick look and maybe even submit a PR if it's a really easy fix.

**Comment by gouchi** [comment]
For the [udev joypad  driver](https://github.com/libretro/RetroArch/blob/master/input/drivers_joypad/udev_joypad.c#L243).

**Comment by gouchi** [comment]
ping @fr500

**Comment by fr500** [comment]
hey, I was pinged but... I don't know a thing about this!
Maybe @twinaphex.

If it works send a PR explaining what's wrong, what this achieves, and what are the caveats.

**Comment by gouchi** [comment]
Hi @auxym did you have the opportunity to provide a PR ?

**Comment by gouchi** [comment]
@auxym Can you make a test with latest RA as this [PR](https://github.com/libretro/RetroArch/pull/12952) has been merged and it should help to fix this issue.

Thank you.

**Comment by gouchi** [comment]
@auxym This issue should be closed. Thank you.
